test/sourcemaps.js

```diff
@@ -26,8 +27,24 @@ describe('sourcemaps', function() {
       .readFileSync(path.join(__dirname, '/dist/index.map'))
       .toString();
     mapValidator(raw, map);
+    let mapObject = JSON.parse(map);
+    assert(
+      mapObject.sourceRoot ===
+        path.relative(b.options.outDir, b.options.rootDir),
+      'sourceRoot should be the root of the source files, relative to the output directory.'
+    );
+    assert(
+      fs.existsSync(
+        path.resolve(
```

Comment:
This is not a sufficient test, because it passes as long as `path.resolve()` can resolve it. But `sourceRoot` is a URL, so you need to test that resolving it as URL results in the right location. Which it currently doesn't, because `sourceRoot` doesn't contain a trailing slash, and trailing slashes are significant in URLs while they don't matter to `path.resolve()`.

```ts
new URL('index.js', 'file:///my/project')  // => 'file:///my/index.js' (WRONG)
new URL('index.js', 'file:///my/project/') // => 'file:///my/project/index.js' (CORRECT)
```

Reply:
See the source map spec:

> #### Resolving Sources
> If the sources are not absolute URLs after prepending of the “sourceRoot”, the sources are resolved relative to the SourceMap (like resolving script src in a html document).

https://sourcemaps.info/spec.html